pvlib/tests/test_tools.py

```diff
@@ -95,3 +96,17 @@ def test_degrees_to_index_1():
     'latitude' or 'longitude' is passed."""
     with pytest.raises(IndexError):  # invalid value for coordinate argument
         tools._degrees_to_index(degrees=22.0, coordinate='width')
+
+
+def test_dataset_passes():
+    expected_dataset = '723170TYA.CSV'
+    assert tools.dataset(expected_dataset).endswith(expected_dataset)
+    assert tools.dataset(pathlib.Path(expected_dataset)) \
+        .endswith(expected_dataset)
```

Comment:
I suggest checking that something exists at the returned file system location

Reply:
Good point.